Dockerfile

```diff
@@ -0,0 +1,18 @@
+# Pull the base image with given version.
+ARG BUILD_TERRAFORM_VERSION=0.11.7
+FROM microsoft/terraform-test:${BUILD_TERRAFORM_VERSION}
```

Comment:
Terraform v12 is so hot right now. Here's another Azure PR to upgrade the test dockerfile https://github.com/Azure/terraform-azurerm-aks/pull/35/files 

I don't mind doing the upgrade in a subsequent PR